compiler/codegen/codegen.cpp

```diff
@@ -2612,15 +2597,29 @@ void codegen() {
   }
 
   codegenPartTwo();
-
 }
 
-//TODO: remove makeBinary
 void makeBinary(void) {
-  // moved to codegenPartTwo
+  if (no_codegen)
+    return;
+
+  if(fLlvmCodegen) {
+#ifdef HAVE_LLVM
+    makeBinaryLLVM();
+#endif
+  } else {
+    const char* makeflags = printSystemCommands ? "-f " : "-s -f ";
+    const char* command = astr(astr(CHPL_MAKE, " "),
+                               makeflags,
+                               getIntermediateDirName(), "/Makefile");
+    mysystem(command, "compiling generated source");
+  }
 
-  // TODO: make runLLVMLinking to here so we can get just 1 executable
-  // Question: where does the GPU kernel go? Is it even in the executable?
+  if (gCodegenGPU == false) {
```

Comment:
Out of curiosity, this probably could be an assert at this point, couldn't it? In other words, can this be true today when we have `fLibraryCompile` and `fLibraryPython` set?

Reply:
I don't think `--library` does anything interesting/meaningful today when using the GPU locale model. I suppose we could try to get a better error for whatever presumably horrible way it fails but it's not a priority.

Reply:
That's reasonable.